201-openshift-origin-rhel/azuredeploy.json

```diff
@@ -0,0 +1,743 @@
+﻿{
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "masterVmSize": {
+      "type": "string",
+      "defaultValue": "Standard_A4",
+      "allowedValues": [
+        "Standard_A4",
+        "Standard_A5",
+        "Standard_A6",
+        "Standard_A7",
+        "Standard_A8",
+        "Standard_A9",
+        "Standard_A10",
+        "Standard_A11",
+        "Standard_D3",
+        "Standard_D4",
+        "Standard_D11",
+        "Standard_D12",
+        "Standard_D13",
+        "Standard_D14",
+        "Standard_D3_v2",
+        "Standard_D4_v2",
+        "Standard_D5_v2",
+        "Standard_D11_v2",
+        "Standard_D12_v2",
+        "Standard_D13_v2",
+        "Standard_D14_v2",
+        "Standard_G1",
+        "Standard_G2",
+        "Standard_G3",
+        "Standard_G4",
+        "Standard_G5"
+      ],
+      "metadata": {
+        "description": "OpenShift Master VM size"
+      }
+    },
+    "nodeVmSize": {
+      "type": "string",
+      "defaultValue": "Standard_A4",
+      "allowedValues": [
+        "Standard_A4",
+        "Standard_A5",
+        "Standard_A6",
+        "Standard_A7",
+        "Standard_A8",
+        "Standard_A9",
+        "Standard_A10",
+        "Standard_A11",
+        "Standard_D3",
+        "Standard_D4",
+        "Standard_D11",
+        "Standard_D12",
+        "Standard_D13",
+        "Standard_D14",
+        "Standard_D3_v2",
+        "Standard_D4_v2",
+        "Standard_D5_v2",
+        "Standard_D11_v2",
+        "Standard_D12_v2",
+        "Standard_D13_v2",
+        "Standard_D14_v2",
+        "Standard_G1",
+        "Standard_G2",
+        "Standard_G3",
+        "Standard_G4",
+        "Standard_G5"
+      ],
+      "metadata": {
+        "description": "OpenShift Node VM(s) size"
+      }
+    },
+    "openshiftMasterHostname": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "OpenShift Master hostname"
+      }
+    },
+    "openshiftMasterPublicIpDnsName": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "OpenShift Master VM public IP DNS name"
+      }
+    },
+    "nodePrefix": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "OpenShift Node VM prefix"
+      }
+    },
+    "nodeInstanceCount": {
+      "type": "int",
+      "minValue": 1,
+	  "allowedValues": [1,2,3,4,5,6,7,8,9,10],
+      "metadata": {
+        "description": "Number of OpenShift nodes"
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "minLength": 1,
+      "metadata": {
+        "description": "Administrator username on all VMs"
+      }
+    },
+    "sshPublicKey": {
+      "type": "securestring",
+      "metadata": {
+        "description": "SSH public key on all VMs"
+      }
+    },
+    "sshPrivateKey": {
```

Comment:
This looks like this is the name of the key vault secret rather than entering the private key here